# PAP DATASET

### List all existing files in "dataset"

In [16]:
import os

base_directory = "/mount/studenten/semantic-plausibility/datasets/pap"

# walkthrough the directory and its subdirs
for root, dirs, files in os.walk(base_directory):
    for file in files:
        print(root)
        print(file)

/mount/studenten/semantic-plausibility/datasets/pap/additional-resources/eichel-ssiw-2023
Amendment.pdf
/mount/studenten/semantic-plausibility/datasets/pap/raw-annotations
README.md
/mount/studenten/semantic-plausibility/datasets/pap/raw-annotations
dataset.tsv
/mount/studenten/semantic-plausibility/datasets/pap/raw-annotations/mace_aggregations
aggregated_predictions.tsv
/mount/studenten/semantic-plausibility/datasets/pap/raw-annotations/mace_aggregations
2-class-dist.prediction
/mount/studenten/semantic-plausibility/datasets/pap/raw-annotations/mace_aggregations
4-class-dist.prediction
/mount/studenten/semantic-plausibility/datasets/pap/train-dev-test-split
README.md
/mount/studenten/semantic-plausibility/datasets/pap/train-dev-test-split/multiclass
train.csv
/mount/studenten/semantic-plausibility/datasets/pap/train-dev-test-split/multiclass
dev.csv
/mount/studenten/semantic-plausibility/datasets/pap/train-dev-test-split/multiclass
test.csv
/mount/studenten/semantic-plausibility/data

### Raw annotation

In [17]:
raw_data_fp = "/mount/studenten/semantic-plausibility/datasets/pap/raw-annotations/dataset.tsv"

In [18]:
import pandas as pd 
raw_df = pd.read_csv(raw_data_fp, sep='\t')


In [19]:
raw_df.head()

event original_label abstractness_combination  \
0          ability means mobility      plausible                    a-m-a   
1  ability permits multiplication      plausible                    a-c-m   
2         access diminishes power      plausible                    a-m-a   
3   achievement reaches community      plausible                    a-c-m   
4        acquisition casts glance      plausible                    a-c-m   

                           rating majority_multiclass  \
0  [2, 5, 4, 5, 5, 2, 5, 5, 5, 5]                   5   
1        [5, 5, 5, 5, 4, 5, 5, 4]                   5   
2     [4, 2, 5, 5, 5, 4, 4, 2, 4]              unsure   
3  [1, 4, 1, 4, 5, 4, 5, 4, 5, 4]              unsure   
4        [4, 5, 4, 4, 4, 4, 1, 2]              unsure   

                             distribution_multiclass majority_binary  \
0                            [0.0, 20.0, 10.0, 70.0]               1   
1                             [0.0, 0.0, 25.0, 75.0]               1   
2  [0.0, 22.22222222222222, 44.44444444444444, 33...               1   
3                            [20.0, 0.0, 50.0, 30.0]               1   
4                           [12.5, 12.5, 62.5, 12.5]               1   

                      distribution_binary  
0                            [80.0, 20.0]  
1                            [100.0, 0.0]  
2  [77.77777777777779, 22.22222222222222]  
3                            [80.0, 20.0]  
4                            [75.0, 25.0]

In [20]:
raw_df.describe()

event original_label abstractness_combination  \
count                     1733           1733                     1733   
unique                    1733              2                       27   
top     ability means mobility    implausible                    m-a-a   
freq                         1            871                       72   

                             rating majority_multiclass  \
count                          1733                1733   
unique                         1708                   2   
top     [5, 5, 5, 4, 4, 5, 4, 5, 5]              unsure   
freq                              3                1667   

       distribution_multiclass majority_binary distribution_binary  
count                     1733            1733                1733  
unique                     409               3                  33  
top     [0.0, 0.0, 50.0, 50.0]               1        [100.0, 0.0]  
freq                        23             931                 159

#### Aspect 1: Basic statistic
* Number of events: 1733 (no missing data)
* Number of original labels: 2 (871 implausible)
* Number of unique abstractness combination: 27
* Most common abstractness combination: m-a-a
* For multiclass, there are a significant amount of unsure class (1677/1733)
* For binary class, there are events rated as plausible (931/1733)
* Number of events where all annotators agree 100%: 159
